doc/man/sphinx-apidoc.rst

```diff
@@ -126,29 +126,253 @@ These options are used when :option:`--full` is specified:
-   sphinx project files generated by apidoc.  Following Jinja2 template
+   sphinx project files generated by apidoc.  The following Jinja2 template
    files are allowed:
 
    * ``module.rst_t``
    * ``package.rst_t``
    * ``toc.rst_t``
+
+   See the section :ref:`apidoctemplating` below for details.
+
+   In addition, when :option:`--full` is specified,
+   :program:`sphinx-quickstart` allows for the following templates:
+
    * ``master_doc.rst_t``
    * ``conf.py_t``
    * ``Makefile_t``
    * ``Makefile.new_t``
    * ``make.bat_t``
    * ``make.bat.new_t``
 
-   In detail, please refer the system template files Sphinx provides.
-   (``sphinx/templates/apidoc`` and ``sphinx/templates/quickstart``)
+   Please refer the to system template files in ``sphinx/templates/quickstart``
+   for details.
+
+   .. versionadded:: 2.2
+
+
+.. _apidoctemplating:
+
+Templating
+----------
+
+.. versionadded:: 2.2
+   Project templating options for sphinx-apidoc
+
+The TOC template
+~~~~~~~~~~~~~~~~
+
+The template for the TOC file, which is generated unless :option:`--no-toc` is
+given, is in the file ``toc.rst_t``. It uses the following Jinja2 variables:
+
+.. data:: header
+
+   The :confval:`project` name, cf. :option:`-H`
+
+.. data:: maxdepth
+
+   Maximum depth for the generated table of contents file, see :option:`-d`
+
+.. data:: docnames
+
+   A sorted list of the modules in the TOC
+
+The default ``toc.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/toc.rst_t
+   :language: jinja
+
+The package template
+~~~~~~~~~~~~~~~~~~~~
+
+The package template (``package.rst_t``) is used to render packages or implicit
+name spaces. It uses the following Jinja2 variables:
+
+.. data:: pkgname
+
+The fully qualified package name
+
+.. data:: subpackages
+
+List of fully qualified sub-package names (if any)
+
+.. data:: submodules
+
+List of fully qualified sub-module names (if any)
+
+.. data:: is_namespace
+
+Whether or not the template is used to render an implicit name space
+
+.. data:: modulefirst
+
+Whether :option:`--module-first` was given
+
+.. data:: separatemodules
+
+Whether :option:`--separate` was given
+
+.. data:: automodule_options
+
+A comma-separated list of ``automodule`` directives, see
+:envvar:`SPHINX_APIDOC_OPTIONS`.
+
+.. data:: show_headings
+
+True unless :option:`--no-headings` was given
+
+
+The default ``package.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/package.rst_t
+   :language: jinja
+
+
+The module template
+~~~~~~~~~~~~~~~~~~~
+
+The module template is used to render modules. This happens only if
+:option:`--separate` was given, or if *MODULE_PATH* contains standalone
+modules instead of packages.
+
+It uses the :data:`automodule_options` and :data:`show_headings` variables of
+``package.rst_t``, and additionally:
+
+.. data:: qualname
+
+The fully qualified name of the module.
+
+.. data:: basename
+
+An alias for :data:`qualname`.
```

Comment:
This documentation matches what is currently implemented. Most likely, it's not what was intended, however. I would assume that `qualname` should be the fully qualified name of the module (with dots), whereas `basename` should only be the part after the last dot. I'll leave fixing this for another PR, though, after this is merged in.

Reply:
The alternative would be for me to change the documentation of `basename` to "The part of the :data:`qualname` after the last dot". This would knowingly create my definition of a "bug" (a mismatch between documentation and behavior). If your preferred definition of "bug" is "mismatch between obvious intent and behavior", on the other hand, we can leave this for now.